File: org/diediedie/Projectile.java
package org.diediedie.actors;
import org.diediedie.actors.Direction;
import org.diediedie.Level;
import org.newdawn.slick.Graphics;
import org.newdawn.slick.Animation;
import org.newdawn.slick.geom.*;

/**
 * Public Interface for all moving weaponary. 
 **/
public interface Projectile extends MovableObject
{
    void updatePosition();
    void updateAiming(float mouseX, float mouseY);
    
    void stop();
    void release(float power);
    
    boolean isFlying();
    float getAngle();
    float getFacingAngle();
    Shape getShape();
    
    float getGravityLine();    
    float getGravity();
    float getMaxGravity();
    
    float getOldStartX();
    float getOldStartY();
    float getDamage();
    
    void increaseGravityEffect();
    
    boolean isGoingDown();
    boolean collidedWithEnemy();
    
    float getEndX();
    float getEndY();    
    float getAirRes();
    
    void calculateEndPos();
    void adjustFacingAngle();
}

File: py/Projectile.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


class Projectile(MovableObject):
    """ generated source for Projectile

    """

    def updatePosition(self):
        raise NotImplementedError()

    def updateAiming(self, mouseX, mouseY):
        raise NotImplementedError()

    def stop(self):
        raise NotImplementedError()

    def release(self, power):
        raise NotImplementedError()

    def isFlying(self):
        raise NotImplementedError()

    def getAngle(self):
        raise NotImplementedError()

    def getShape(self):
        raise NotImplementedError()

    def getGravityLine(self):
        raise NotImplementedError()

    def getGravity(self):
        raise NotImplementedError()

    def getMaxGravity(self):
        raise NotImplementedError()

    def increaseGravityEffect(self):
        raise NotImplementedError()

    def isGoingDown(self):
        raise NotImplementedError()

    def getEndX(self):
        raise NotImplementedError()

    def getEndY(self):
        raise NotImplementedError()

    def getAirRes(self):
        raise NotImplementedError()

    def calculateEndPos(self):
        raise NotImplementedError()

    def adjustFacingAngle(self):
        raise NotImplementedError()


